src/render/src/encoder.rs

```diff
@@ -1,19 +1,91 @@
 //! Graphics commands encoder.
 
-#![deny(missing_docs)]
-
 use draw_state::target::{Depth, Stencil};
 use std::error::Error;
 use std::any::Any;
 use std::{fmt, mem};
+use std::cell::UnsafeCell;
+use std::sync::{mpsc, Arc};
 
-use core::{Backend, CommandQueue, GraphicsCommandPool, GraphicsQueue,
+use core::{Backend, CommandQueue, CommandPool,
            IndexType, SubmissionResult, VertexCount};
-use core::{self, command, format, handle, texture};
-use core::command::{GraphicsCommandBuffer, Submit};
-use core::memory::{self, cast_slice, Typed, Pod, Usage};
-use slice;
-use pso;
+use core::{self, command, format};
+use core::command::CommandBuffer;
+use memory::{self, cast_slice, Typed, Pod, Usage};
+use {handle, image};
+
+// This is the unique owner of the inner UnsafeCell.
+pub struct Scope<B: Backend, C>(Arc<UnsafeCell<ScopeInner<B, C>>>);
```

Comment:
could use `Arc<Mutex<_>>` instead. it's safer and is fast with `futex` backing when no contention is present

Reply:
If you want, but there is not even dynamic ownership.

Reply:
I don't strongly mind either way, it just seems that having less `unsafe` in the codebase is slightly more appealing, if there is no cost to it